pinot-spi/src/main/java/org/apache/pinot/spi/stream/PartitionGroupStatus.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.stream;
+
+/**
+ * A PartitionGroup is a group of partitions/shards that the same consumer should consume from.
+ * This class contains all information which describes the latest state of a partition group.
+ * It is constructed by looking at the segment zk metadata of the latest segment of each partition group.
+ * It consists of:
+ * 1. partitionGroupId - A unique ID for the partitionGroup
+ * 2. sequenceNumber - The sequenceNumber this partitionGroup is currently at
+ * 3. startOffset - The start offset that the latest segment started consuming from
+ * 4. endOffset - The endOffset (if segment consuming from this partition group has finished consuming the segment and recorded the end offset)
+ * 5. status - IN_PROGRESS/DONE
+ *
+ * This information is needed by the stream, when grouping the partitions/shards into new partition groups.
+ */
+public class PartitionGroupStatus {
```

Comment:
I thought we agreed on `PartitionGroupConsumptionStatus`

Reply:
oops my bad. Changed this.